webapp/src/main/java/com/box/l10n/mojito/rest/textunit/TextUnitWS.java

```diff
@@ -87,6 +90,10 @@ public class TextUnitWS {
 
   @Autowired AssetRepository assetRepository;
 
+  @Autowired SlackWarningMessageSender slackWarningMessageSender;
+
+  @Autowired TMTextUnitRepository tmTextUnitRepository;
```

Comment:
As mentioned in another comment you can inject this into the `SlackWarningMessageSender` and do the lookup there